### PR TITLE
feat: deactivate use of meta-transaction when the connected wallet is…

### PR DESCRIPTION
**Pull request by levalleux-ludo**
… a contract

## Description

related to https://github.com/bosonprotocol/interface/issues/1026

- Add isSignerContract() method in Web3LibAdapter interface
- Implement isSignerContract() method in EthersAdapter
- mock isSignerContract() method in EthConnectAdapter (always return false)
- Use isSignerContract() everywhere required to chose between meta-transactions and direct transactions

## How to test

There is no way to create a Gnosis Safe Multisig on testnet (Mumbai or Goerli)
I've tested the feature with my personal Multisig on Polygon / usecase: creating a Seller in the dApp on production env (I didn't validate the transaction)
Other usecases have not been tested, though :-(

### CODE REVIEW

File: packages/react-kit/src/components/cta/offer/CreateOfferButton.tsx [hunk 1]
```diff
@@ -50,7 +50,9 @@ export const CreateOfferButton = ({
       try {
         setIsLoading(true);
         onPendingSignature?.();
-        const isMetaTx = Boolean(coreSdk.isMetaTxConfigSet && signerAddress);
+        const isMetaTx = Boolean(
+          coreSdk.isMetaTxConfigSet && signerAddress && !signerContract
```
Comment (albertfolch-redeemeum) [inline packages/react-kit/src/components/cta/offer/CreateOfferButton.tsx:54]: I feel we are repeating this logic everywhere, could we do something about this?

**Review by albertfolch-redeemeum — APPROVED** [review]

File: packages/react-kit/src/components/cta/offer/CreateOfferButton.tsx (hunk 1 above)
Comment (levalleux-ludo) [reply packages/react-kit/src/components/cta/offer/CreateOfferButton.tsx:54]: Interesting idea.
However, not sure how to handle that:
- coreSdk.isMetaTxConfigSet is a synchnronous check based on configuration
- while signerAddress and signerContract are async and depends on user actions (Wallet connection)

Comment (albertfolch-redeemeum) [reply packages/react-kit/src/components/cta/offer/CreateOfferButton.tsx:54]: let's add a TODO so we dont forget about this improvement and we dont block this PR

Comment (levalleux-ludo) [reply packages/react-kit/src/components/cta/offer/CreateOfferButton.tsx]: done

**Review by albertfolch-redeemeum — APPROVED** [review]

**Pull request merged**